part/txn.go

```diff
@@ -519,12 +527,18 @@ func (txn *Txn[T]) removeChild(parent *header[T], index int) (newParent *header[
 			remainingIndex = 1
 		}
 
+		if parent.watch != nil {
+			txn.watches[parent.watch] = struct{}{}
+		}
+
 		child := parent.node4().children[remainingIndex]
 		// Clone for prefix adjustment, but leave watch alone.
+		// The node must not be marked as mutated since we didn't clone
```

Comment:
We could in principle do some more bookkeeping in `nodeMutated` to know whether on `txn.cloneNode` we should swap the watch or not. But I think it's not worth it as this case is rare enough.